### PR TITLE
Added -Scope to the Resource to fix bugs with narrower scope always m…

### PR DESCRIPTION
**Pull request by HansOMartinsen**
…aking Test-TargetResource returning false.

Signed-off-by: Hans O. Martinsen

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/xpowershellexecutionpolicy/9)

### CODE REVIEW

**Comment by tommyholen** [comment]
Reviewed 3 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpowershellexecutionpolicy/9)*

File: Tests/Unit/MSFT_xPowerShellExecutionpolicy.test.ps1 [hunk 1]
```diff
@@ -54,6 +59,16 @@ try
 
                 $result.ExecutionPolicy | should be $(Get-ExecutionPolicy)
             } 
+
+            It 'Throws when passed an invalid execution policy Scope' {
+                { Get-TargetResource -ExecutionPolicy $(Get-ExecutionPolicy) -Scope "badParam" } | should throw $invalidPolicyScopeThrowMessage
+            }
+
+            It 'Returns correct execution policy for the correct Scope' {
+                $result = Get-TargetResource -ExecutionPolicy $(Get-ExecutionPolicy) -Scope 'LocalMachine'
```
Comment (BerheAbrha) [inline Tests/Unit/MSFT_xPowerShellExecutionpolicy.test.ps1]: >ExecutionPolicy [](start = 68, length = 15)

I think you need to specify the parameter  -Scope 'LocalMachine'  in this and the line below.

File: Tests/Unit/MSFT_xPowerShellExecutionpolicy.test.ps1 [hunk 2]
```diff
@@ -54,6 +59,16 @@ try
...
+                $result.ExecutionPolicy | should be $(Get-ExecutionPolicy)
```
Comment (BerheAbrha) [inline Tests/Unit/MSFT_xPowerShellExecutionpolicy.test.ps1]: >Get-ExecutionPolicy [](start = 54, length = 19)

Same comment as above.

**Comment by BerheAbrha** [comment]
:clock1:

**Comment by johlju** [comment]
@HansOMartinsen are you able to work on this PR if I review? If so could you start by rebasing (git rebase) this PR to resolve the conflict? If you can’t work on this then let us know and someone else could continue the work. Thanks!

**Comment by johlju** [comment]
Closing this as this is being worked on in PR #15

**Pull request closed without merging**